skxray/img_proc/mathops.py

```diff
@@ -0,0 +1,349 @@
+                 total equal to the left operand,
+                 e.g.:
+                    b+=a is equivalent to b=a+b
+            -= : Subtracts the right operand from the left operand and sets
+                 the total equal to the left operand,
+                 e.g.:
+                    b -= a is equivalent to b = b - a
+            *= : multiplies the right operand to the left operand and sets the
+                 total equal to the left operand,
+                 e.g.:
+                    b *= a is equivalent to b = b * a
+            /= : divides the right operand into the left operand and sets the
+                 total equal to the left operand,
+                 e.g.:
+                    b /= a is equivalent to b = b / a
+            %= : divides the right operand into the left operand and sets the
+                 remainder equal to the left operand,
+                 e.g.:
+                    b %= a is equivalent to b =b % a
+            **= : raises the left operand to the power of the right operand
+                  and sets the total equal to the left operand,
+                  e.g.:
+                    b **= a is equivalent to b = b ** a
+            //= : divides the right operand into the left operand and
+                  then removes any values after the decimal point. The total
+                  is then set equal to the left operand,
+                 e.g.:
+                    b //= a is equivalent to b = b // a
+
+        In the event that bitwise operations are required the operators &,
+        |, ^, ~ may also be used, though I'm struggling to come up with a
+        scenario where this will be used.
+
+        Order of operations and parenthesis are taken into account when
+        evaluating the expression.
+
+    A : {ndarray, int, float}
+        Data set or constant to be offset or manipulated
+
+    B : {ndarray, int, float}
+        Data set or constant to be offset or manipulated
+
+    C : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    D : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    E : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    F : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    G : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+    H : {ndarray, int, float}, optional
+        Data set or constant to be offset or manipulated
+
+
+    Returns
+    -------
+    output : {ndarray, int, float}
+        Returns the resulting array or value to the designated variable
+
+    Example
+    -------
+    result = mathops.arithmetic_custom('(A+C)/(B+D)', img_1, img_2, 2, 4)
+    """
+    output = eval(parser.expr(expression).compile())
+    return output
+
+
+def logic_basic(operation,
+                src_data1,
+                src_data2=None):

```

Comment:
Completed for both arithmetic_basic, and logic_basic. In addition, the arg order for all of the functions has been standardized to start with the operation or expression definition, followed by the input data.